### PR TITLE
test: remove cross-project fleet workload identity tests

### PR DESCRIPTION
**Pull request by nan-yu**
The fleet team added a registration guardrail that forces all cluster projects to be registered to at most one fleet. The cross-project testcases register the CI clusters to both the same project and a cross-project, cs-dev-hub. This commit removes the cross-project testcases because they are covered in platform qualification tests.

Reference: b/364363769

### CODE REVIEW

File: Makefile.oss.prow [hunk 1]
```diff
@@ -81,11 +81,3 @@ KCC_MANAGED_PROJECT ?= cs-dev-hub
 set-up-kcc-configs:
```
[the remark this answers is not in the document]
Comment (nan-yu) [reply Makefile.oss.prow:81]: False positive: these variables are already removed in the commit.

File: e2e/nomostest/testutils/fleet_membership.go [hunk 2]
```diff
@@ -29,9 +29,6 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// TestCrossProjectFleetProjectID is the fleet host project.
-const TestCrossProjectFleetProjectID = "cs-dev-hub"
-
 // RegisterCluster registers a cluster in a fleet.
```
[the remark this answers is not in the document]
Comment (nan-yu) [reply e2e/nomostest/testutils/fleet_membership.go:32]: False positive: This change is to remove the variable.

File: e2e/testcases/workload_identity_test.go [hunk 3]
```diff
@@ -314,21 +242,16 @@ func TestWorkloadIdentity(t *testing.T) {
...
 
 			rootSync := k8sobjects.RootSyncObjectV1Beta1(rootSyncID.Name)
 			repoSync := k8sobjects.RepoSyncObjectV1Beta1(repoSyncID.Namespace, repoSyncID.Name)
 			nt.T.Cleanup(func() {
 				testutils.ClearMembershipInfo(nt, fleetMembership, *e2e.GCPProject, gkeURI)
-				testutils.ClearMembershipInfo(nt, fleetMembership, testutils.TestCrossProjectFleetProjectID, gkeURI)
 			})
 
 			// Register the cluster for fleet workload identity test
 			if tc.fleetWITest {
```
[the remark this answers is not in the document]
Comment (nan-yu) [reply e2e/testcases/workload_identity_test.go:253]: False comment: The conditional check `tc.fleetWITest` is needed to conditionally register the cluster.

File: e2e/testinfra/README.md [hunk 4]
```diff
@@ -5,7 +5,6 @@ Most GCP resources are managed by [Terraform](./terraform/README.md), but the
 following resources need to be configured separately:
 - Test OCI images on Google Container Registry
 - KCC configurations
```
[the remark this answers is not in the document]
Comment (nan-yu) [reply e2e/testinfra/README.md:7]: False positive: I didn't get the change. The README is already updated to reflect the change.

File: e2e/testinfra/README.md [hunk 5]
```diff
@@ -21,8 +20,6 @@ Below is a list of environment variables required by the setup scripts:
 prow jobs. The default value is `oss-prow-build-kpt-config-sync`.
```
[the remark this answers is not in the document]
Comment (nan-yu) [reply e2e/testinfra/README.md:20]: False positive: it is already removed and no longer mentioned in the project.

File: e2e/testinfra/README.md [hunk 6]
```diff
@@ -34,8 +31,3 @@ value is `cs-dev-hub`.
     ```bash
```
[the remark this answers is not in the document]
Comment (nan-yu) [reply e2e/testinfra/README.md:31]: False positive: It is already removed.

File: e2e/testcases/workload_identity_test.go [hunk 7]
```diff
@@ -141,35 +126,6 @@ func TestWorkloadIdentity(t *testing.T) {
...
-			nsSrcCfg:         sourceConfig{pkg: "hydration/namespace-repo", dir: ".", version: "v1"},
-			newRootSrcCfg:    sourceConfig{pkg: "hydration/kustomize-components", dir: "tenant-a", version: "v1"},
-			newNSSrcCfg:      sourceConfig{pkg: "hydration/namespace-repo", dir: "test-ns", version: "v1"},
-			sourceType:       configsync.OciSource,
-			gsaEmail:         gsaARReaderEmail(),
-			testKSAMigration: true,
-			requireOCIGAR:    true,
-		},
-		{
-			name:         "Authenticate to OCI image on GCR with Fleet WI in the same project",
```
Comment (sdowell) [inline e2e/testcases/workload_identity_test.go:155]: Why were these test cases removed?

Comment (sdowell) [reply e2e/testcases/workload_identity_test.go:155]: Is the diff just showing up funny because the `crossProject` bit was flipped on the test cases below?

Comment (nan-yu) [reply e2e/testcases/workload_identity_test.go:155]: These tests are still there. I think the diff is showing in a confusing way. Some fields were also removed, which causes reformatting for the test table.

File: scripts/set-up-workload-identity-configs.sh [hunk 8]
```diff
@@ -1,78 +0,0 @@
...
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-set -euo pipefail
-
-# This script sets up the test-infra for validating the authentication via workload identity.
```
Comment (sdowell) [inline scripts/set-up-workload-identity-configs.sh:18]: Are you planning to deprovision these resources?

**Review by sdowell — APPROVED** [review]
/lgtm

File: scripts/set-up-workload-identity-configs.sh (hunk 8 above)
Comment (nan-yu) [reply scripts/set-up-workload-identity-configs.sh:18]: Yes, I'll.

**Pull request merged**